### ISSUE TITLE
`conf.Swarm.AddrFilters` are not set in `~/.ipfs/config` if repo is already init'ed

### ISSUE THREAD

**Issue by altergui** [vocdoni/vocdoni-node#450]
@vdo reported that ipfsSync is triggering scan alerts in staging.

ipfs node has this filter to avoid this
https://github.com/vocdoni/vocdoni-node/blob/stage/ipfs/ipfsInit.go#L65

so i suspect that the subpub (libp2p) node is doing these.

will dig into it next week

**Comment by altergui** [comment]
so, it turns out that this particular staging node (who was triggering the alerts) had an outdated ~/.ipfs/config which did not specify conf.Swarm.AddrFilters
@vdo fixed this in https://github.com/vocdoni/vocdoni-node/blob/ee4021e9c7c77445740ef84498c10f8476fb1a0f/ipfs/ipfsInit.go#L63-L82
but that code is only run during ipfsInit and this happens only once: if the host already has an `~/.ipfs` then `dvotenode` will not re-init (reconfigure) it.

**Comment by altergui** [comment]
it probably makes sense to make https://github.com/vocdoni/vocdoni-node/blob/ee4021e9c7c77445740ef84498c10f8476fb1a0f/ipfs/ipfsInit.go#L36
idempotent and run this unconditionally on every `dvotenode` start.

right now, it's checked twice if the repo exists and in that case doesn't touch (update) the config
https://github.com/vocdoni/vocdoni-node/blob/ee4021e9c7c77445740ef84498c10f8476fb1a0f/data/ipfs.go#L62-L67
https://github.com/vocdoni/vocdoni-node/blob/ee4021e9c7c77445740ef84498c10f8476fb1a0f/ipfs/ipfsInit.go#L43-L45

**Comment by altergui** [comment]
the last people touching this code were @NateWilliams2 and @p4u, if you have opinions please weigh in, otherwise i'd work on a PR to enforce the ipfs config on every run (so that also any config changes we might do in the future will also impact the preexisting `~/.ipfs/config`)

**Comment by jordipainan** [comment]
@altergui any updates on this ?

**Comment by altergui** [comment]
thanks for the reminder! will prepare a trivial PR

**Comment by altergui** [comment]
well... it's not so trivial as i expected. 
upstream code also checks for IsInitialized
https://github.com/ipfs/go-ipfs/blob/eafa1c73455d91850da61f09068af4535cf499e6/cmd/ipfs/init.go#L147-L159

and it has no concept of "loading" a config and modifying it in place. instead, it has code for doing the init based on some preexisting config file
https://github.com/ipfs/go-ipfs/blob/eafa1c73455d91850da61f09068af4535cf499e6/cmd/ipfs/init.go#L97
it does look as simple as doing that `json.NewDecoder(file).Decode(conf)` during our init, but i have the feeling this will take more development time than expected, and right now i prefer to mark this issue as WONTFIX, and only work on this WHEN we make a change to ipfs config in code, that we want to be applied on top of already-inited configs.
